test/functional/rpc_setban.py

```diff
@@ -51,12 +51,24 @@ def run_test(self):
         ip_addr = "1.2.3.4"
         assert(not self.is_banned(node, tor_addr))
         assert(not self.is_banned(node, ip_addr))
+
         node.setban(tor_addr, "add")
         assert(self.is_banned(node, tor_addr))
         assert(not self.is_banned(node, ip_addr))
+
+        # Test that the ban list is preserved through restart.
+
+        self.restart_node(1)
+        assert(self.is_banned(node, tor_addr))
+        assert(not self.is_banned(node, ip_addr))
+
         node.setban(tor_addr, "remove")
         assert(not self.is_banned(self.nodes[1], tor_addr))
         assert(not self.is_banned(node, ip_addr))
 
+        self.restart_node(1)
+        assert(not self.is_banned(self.nodes[1], tor_addr))
+        assert(not self.is_banned(node, ip_addr))
```

Comment:
Done. Since the surrounding code uses `assert()` I added the new code with `assert()` for consistency and then removed all `()` from asserts in the entire file in a separate commit.